src/get-started/java-binding.md

```diff
@@ -491,75 +493,201 @@ as specified in [How to Run a Service section](#how-to-run-a-service).
+[transaction messages][transactions-messages]:
+
+```java
+try (TestKit testKit = TestKit.forService(MyServiceModule.class)) {
+  // Construct some transaction to be executed
+  TransactionMessage message = constructTransactionMessage();
+  // Commit block with this transaction
+  Block block = testKit.createBlockWithTransactions(message);
+  // Check the resulting block or blockchain state
+}
+```
+
+TestKit also allows creating blocks that contain all current [in-pool][in-pool]
+transactions:
+
+```java
+try (TestKit testKit = TestKit.forService(MyServiceModule.class)) {
+  // Put the transaction into TestKit transaction pool
+  MyService service = testKit.getService(MyService.SERVICE_ID, MyService.class);
+
+  TransactionMessage message = constructTransactionMessage();
+  RawTransaction rawTransaction = RawTransaction.fromMessage(message);
+  service.getNode().submitTransaction(rawTransaction);
 
-### Read Requests
+  // This block will contain transaction submitted above
+  Block block = testKit.createBlock();
+  // Check the resulting block or blockchain state
+}
+```
+
+TestKit provides [`getTransactionPool()`][testkit-get-pool] and
+[`findTransactionsInPool(Predicate<TransactionMessage> predicate)`][testkit-find-in-pool]
+methods to inspect transaction pool. It is useful when there is a need to verify
+transactions submitted in `afterCommit` method, as those are put into the pool.
+
+!!! note
+    Note that blocks that are created with
+    [`TestKit.createBlockWithTransactions(Iterable<TransactionMessage> transactionMessages)`][testkit-create-block]
+    will ignore in-pool transactions. As of 0.7.0, there is no way to create a block
+    that would contain both given and in-pool transactions with a single
+    method - to do that, put given transactions into TestKit transaction pool
+    with [`Node.submitTransaction(RawTransaction rawTransaction)`][node-submit-transaction].
+
+#### Checking the blockchain state
+
+In order to verify changes in blockchain state TestKit provides a snapshot of
+the current database state (i.e., the one that corresponds to the latest
+committed block). There are several ways to access it:
+
+- `void withSnapshot(Consumer<Snapshot> snapshotFunction)`
+  Performs a given function with a snapshot of the current database state
+- `<ResultT> ResultT applySnapshot(Function<Snapshot, ResultT> snapshotFunction)`
+  Performs a given function with a snapshot of the current database state and
+  returns a result of its execution
+- `Snapshot getSnapshot()`
+  Returns a snapshot of the current database state
+
+!!! note
+    Note that `withSnapshot` and `applySnapshot` methods destroy the snapshot
+    once the passed closure completes, compared to `getSnapshot`, which
+    disposes created snapshots only when TestKit is closed.
+    Therefore it is recommended to use first two methods if a large number
+    (e.g. more than a hundred) of snapshots needs to be created.
```

Comment:
Example with `getSnapshot` and a simple service (e.g., "Suppose there is a service
with a single transaction incrementing a counter stored in `EntryIndex`")?

Or we might leave it out if we add it to the first example in "Transactions Testing" section.